lib/smart_city/event/dataset_update.ex

```diff
@@ -0,0 +1,149 @@
+defmodule SmartCity.Event.DatasetUpdate do
+  @moduledoc """
+  Struct defining a dataset update event.  This is triggered when new datasets are put into the system or existing
+  datasets are updated
+
+  ```javascript
+  const Dataset = {
+    "id": "",                  // UUID
+    "business": {              // Project Open Data Metadata Schema v1.1
+      "dataTitle": "",       // user friendly (dataTitle)
+      "description": "",
+      "keywords": [""],
+      "modifiedDate": "",
+      "orgTitle": "",        // user friendly (orgTitle)
+      "contactName": "",
+      "contactEmail": "",
+      "license": "",
+      "rights": "",
+      "homepage": "",
+      "spatial": "",
+      "temporal": "",
+      "publishFrequency": "",
+      "conformsToUri": "",
+      "describedByUrl": "",
+      "describedByMimeType": "",
+      "parentDataset": "",
+      "issuedDate": "",
+      "language": "",
+      "referenceUrls": [""],
+      "categories": [""]
+    },
+    "technical": {
+      "dataName": "",        // ~r/[a-zA-Z_]+$/
+      "orgId": "",
+      "orgName": "",         // ~r/[a-zA-Z_]+$/
+      "systemName": "",      // ${orgName}__${dataName},
+      "schema": [
+        {
+          "name": "",
+          "type": "",
+          "description": ""
+        }
+      ],
+      "sourceUrl": "",
+      "protocol": "",       // List of protocols to use. Defaults to nil. Can be [http1, http2]
+      "authUrl": "",
+      "sourceFormat": "",
+      "sourceType": "",     // remote|stream|ingest|host
+      "cadence": "",
+      "sourceQueryParams": {
+        "key1": "",
+        "key2": ""
+      },
+      "transformations": [], // ?
+      "validations": [],     // ?
+      "sourceHeaders": {
+        "header1": "",
+        "header2": ""
+      }
+      "authHeaders": {
+        "header1": "",
+        "header2": ""
+      }
+    },
+    "_metadata": {
+      "intendedUse": [],
+      "expectedBenefit": []
+    }
+  }
+  ```
+  """
+
+  alias SmartCity.Event.DatasetUpdate.Business
+  alias SmartCity.Event.DatasetUpdate.Technical
+  alias SmartCity.Event.DatasetUpdate.Metadata
+
+  @type id :: term()
+  @type t :: %SmartCity.Event.DatasetUpdate{
```

Comment:
This is a big, tedious ask, but I figure I'll be bashful.
Would you guys be willing to alphabetize all our struct fields? Currently our types/documentation/defstructs seem to have no order at all, which I think makes scanning/browsing a lot slower for users of the library.

Reply:
done

Reply:
Wow, thank you guys for taking this hit. I appreciate it!